### PR TITLE
[change] filtering out transitive dependencies sources that are also root targets

### PR DESCRIPTION
**Pull request by sellophane**

### CODE REVIEW

**Review by abrams27 — CHANGES_REQUESTED** [review]
tests are failing because we are expecting these sources now - just remove them from tests

**Review by abrams27 — APPROVED** [review]
please update the changelog + i think that we can get rid of `sourcesClasspath` field - @lukaszwawrzyk what do u thing?

File: server/src/main/java/org/jetbrains/bsp/bazel/server/sync/languages/java/JavaLanguagePlugin.kt [hunk 1]
```diff
@@ -76,10 +78,22 @@ class JavaLanguagePlugin(
...
+        targetInfo.getJavaTargetInfoOrNull()?.run {
+            dependencyTree.transitiveDependenciesWithoutRootTargets(targetInfo.id)
+                .flatMap(::getSourceJars)
+                .map(bazelPathsResolver::resolveUri)
+                .toSet()
         }.orEmpty()
 
+    private fun getSourceJars(targetInfo: TargetInfo): List<FileLocation> =
+        targetInfo.getJavaTargetInfoOrNull()
+            ?.jarsOrBuilderList
```
[the remark this answers is not in the document]
Comment (abrams27) [reply server/src/main/java/org/jetbrains/bsp/bazel/server/sync/languages/java/JavaLanguagePlugin.kt:90]: so how can we fix it?

[the remark this answers is not in the document]
Comment (lukaszwawrzyk) [reply server/src/main/java/org/jetbrains/bsp/bazel/server/sync/languages/java/JavaLanguagePlugin.kt:90]: Just use another field and concat result. Will happen on a followup pr.

**Pull request merged**